### DISCUSSION TITLE
How to add an append/add column button after the last column?

### DISCUSSION THREAD

**Discussion by jorenrui** [glideapps/glide-data-grid#456] [Q&A] [answered]
Hello, I was wondering if there is a way to add a "add column" button after the last column or will it be implemented in the future? Like how "row append" works.

![image](https://user-images.githubusercontent.com/23741509/182549169-3023dfef-f50b-4a02-b8ee-bc1081d7607f.png)

Currently, the only way I add new columns is by having something like this:

![image](https://user-images.githubusercontent.com/23741509/182549536-3cb33ba6-1f75-471c-8f24-97e764079ec9.png)

And using the [rightElement](https://github.com/glideapps/glide-data-grid/blob/main/packages/core/API.md#rightelement) to add a "sticky" add column like this:

![image](https://user-images.githubusercontent.com/23741509/182549716-d9029fbe-a9fd-42f9-ae15-ecc06c7fd256.png)

I was trying to create something like this in Airtable:

![image](https://user-images.githubusercontent.com/23741509/182550375-ab81c996-d72a-408f-a40b-eb459623ed95.png)

Currently, another solution that I think of is just adding a column that draws a custom header cell with a plus icon (or plus header icon with no label). Then add a [onHeaderClicked](https://github.com/glideapps/glide-data-grid/blob/main/packages/core/API.md#onheaderclicked) for it.

Anyways, I really appreciate the data grid you're building. Love how fast and smooth it is!

**Comment by jassmith** [comment]
Let me try my hand at making the airtable style approach.

**Comment by jassmith** [comment, marked as answer]
I have added an example here: https://github.com/glideapps/glide-data-grid/pull/416/commits/0767cc1beba6f627dfef2746026f28f805747bf6

Note that this commit includes a breaking change which is not needed for the effect to work, it just adds more flexibility for me in the future to make these right element items more flexible. In the current API you merely need to make sure at least 1 column has `grow: 1` set and that rightElementSticky is false. Then you can implement the button using the correct theme values (I just hardcoded the correct values because demo code).

At the end you get something that looks like this:

![image](https://user-images.githubusercontent.com/30443/182639325-264f4296-a475-4da5-be86-ef1a8df1bab6.png)

**Reply by jassmith** [reply]
Making this a default feature could be nice, I guess I never really thought about that. I dont want to bloat the package but I suspect this is a pretty common use case. It could probably be done as a third party hook package.

**Reply by jorenrui** [reply]
Okiee. Thank you, I appreciate this! 😁
